src/main/java/games/strategy/engine/lobby/server/userDB/DerbyUserController.java

```diff
@@ -69,71 +76,72 @@ public boolean doesUserExist(final String userName) {
   }
 
   @Override
-  public void updateUser(final String name, final String email, final String hashedPassword, final boolean admin) {
-    final String validationErrors = UserDao.validate(name, email, hashedPassword);
-    if (validationErrors != null) {
-      throw new IllegalStateException(validationErrors);
-    }
-    final Connection con = Database.getConnection();
+  public void updateUser(final DbUser user, final HashedPassword hashedPassword) {
+    Preconditions.checkArgument(user.isValid(), user.getValidationErrorMessage());
+
+    final Connection con = Database.getDerbyConnection();
```

Comment:
Should this be using `connectionSupplier.get()` instead of `Database.getDerbyConnection()`?